buffer/src/test/java/io/netty/buffer/ReadOnlyByteBufTest.java

```diff
@@ -182,6 +193,28 @@ public void shouldIndicateNotWritableAnyNumber() {
     }
 
     @Test
+    public void ensureWritableIntStatusShouldFailButNotThrow() {
+        ByteBuf buf = buffer(1);
+        ByteBuf readOnly = buf.asReadOnly();
+        int result = unmodifiableBuffer(readOnly).ensureWritable(1, false);
```

Comment:
why wrap in `unmodifiableBuffer` ? you called as `asReadOnly()` before so its already a read-only buffer.